airgap/terraform/main.tf

```diff
@@ -0,0 +1,309 @@
+
+  tags = {
+    Name = "lh-registry-public-subnet-${random_string.random_suffix.id}"
+  }
+}
+
+# Create private subnet
+resource "aws_subnet" "lh_registry_aws_private_subnet" {
+  vpc_id     = aws_vpc.lh_registry_aws_vpc.id
+  availability_zone = data.aws_availability_zones.available.names[1]
+  cidr_block = "10.0.2.0/24"
+
+  map_public_ip_on_launch = true
+
+  tags = {
+    Name = "lh-registry-private-subnet-${random_string.random_suffix.id}"
+  }
+}
+
+# Create route table for public subnets
+resource "aws_route_table" "lh_registry_aws_public_rt" {
+  depends_on = [
+    aws_internet_gateway.lh_registry_aws_igw,
+  ]
+
+  vpc_id = aws_vpc.lh_registry_aws_vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.lh_registry_aws_igw.id
+  }
+
+  tags = {
+    Name = "lh-registry-aws-public-rt-${random_string.random_suffix.id}"
+  }
+}
+
+# Create route table for private subnets
+resource "aws_route_table" "lh_registry_aws_private_rt" {
+  depends_on = [
+    aws_internet_gateway.lh_registry_aws_igw
+  ]
+
+  vpc_id = aws_vpc.lh_registry_aws_vpc.id
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.lh_registry_aws_igw.id
+  }
+
+  tags = {
+    Name = "lh-registry-aws-private-rt-${random_string.random_suffix.id}"
+  }
+}
+
+# Associate public subnet to public route table
+resource "aws_route_table_association" "lh_registry_aws_public_subnet_rt_association" {
+  depends_on = [
+    aws_subnet.lh_registry_aws_public_subnet,
+    aws_route_table.lh_registry_aws_public_rt
+  ]
+
+  subnet_id      = aws_subnet.lh_registry_aws_public_subnet.id
+  route_table_id = aws_route_table.lh_registry_aws_public_rt.id
+}
+
+# Associate private subnet to private route table
+resource "aws_route_table_association" "lh_registry_aws_private_subnet_rt_association" {
+  depends_on = [
+    aws_subnet.lh_registry_aws_private_subnet,
+    aws_route_table.lh_registry_aws_private_rt
+  ]
+
+  subnet_id      = aws_subnet.lh_registry_aws_private_subnet.id
+  route_table_id = aws_route_table.lh_registry_aws_private_rt.id
+}
+
+# Create AWS key pair
+resource "aws_key_pair" "lh_registry_aws_pair_key" {
+  key_name   = format("%s_%s", "lh_registry_aws_key_pair", random_string.random_suffix.id)
+  public_key = file(var.aws_ssh_public_key_file_path)
+}
+
+# Create instance for registry
+resource "aws_instance" "lh_registry_aws_instance" {
+
+  depends_on = [
+    aws_subnet.lh_registry_aws_public_subnet,
+  ]
+
+  availability_zone = data.aws_availability_zones.available.names[0]
+
+  ami           = data.aws_ami.aws_ami_ubuntu.id
+  instance_type = var.registry_aws_instance_type
+  associate_public_ip_address = true
+
+  subnet_id = aws_subnet.lh_registry_aws_public_subnet.id
+  vpc_security_group_ids = [
+    aws_security_group.lh_registry_aws_secgrp.id
+  ]
+
+  key_name = aws_key_pair.lh_registry_aws_pair_key.key_name
+
+  user_data = data.template_file.provision_registry_server.rendered
```

Comment:
The content in `provision_registry_server` is not really about the register server provisioning, so probably the name can be changed to a meaningful name?

Reply:
Remove provision_registry_server template file, and rename script.sh to provision_registry_server.sh, then test again.